File: admin-be/src/main/java/com/bsep/admin/myHouse/dto/Report.java
package com.bsep.admin.myHouse.dto;

import com.bsep.admin.model.DeviceType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Report {
    DeviceType deviceType;
    String deviceName;
    int numberOfAlarmMessages;
    int numberOfInfoMessages;
    int numberOfMessages;
    Double minValue;
    Double maxValue;
    Double averageValue;
}

File: user-fe/src/app/model/report.ts
import { DeviceType } from "./myhouse";

export class Report {
    deviceType: DeviceType = DeviceType.THERMOMETER;
    deviceName: string = '';
    numberOfAlarmMessages: number = 0;
    numberOfInfoMessages: number = 0;
    numberOfMessages: number = 0;
    minValue: number = 0;
    maxValue: number = 0;
    averageValue: number = 0;
}